File: apollo-compiler/src/test/graphql/com/example/input_object_type/type/CustomType.java
package com.example.input_object_type.type;

import com.apollographql.apollo.api.ScalarType;
import java.lang.Class;
import java.lang.Override;
import java.lang.String;
import java.util.Date;
import javax.annotation.Generated;

@Generated("Apollo GraphQL")
public enum CustomType implements ScalarType {
  DATE {
    @Override
    public String typeName() {
      return "Date";
    }

    @Override
    public Class javaType() {
      return Date.class;
    }
  }
}

File: apollo-compiler/src/test/graphql/com/example/input_object_type/type/CustomType.kt
package com.example.input_object_type.type

import com.apollographql.apollo.api.ScalarType
import java.lang.Class
import javax.annotation.Generated
import kotlin.String

@Generated("Apollo GraphQL")
enum class CustomType : ScalarType {
    DATE {
        override fun typeName(): String = "Date"

        override fun javaType(): Class<*> = java.util.Date::class.java
    },

    ID {
        override fun typeName(): String = "ID"

        override fun javaType(): Class<*> = java.lang.Integer::class.java
    }
}
